src/Switch.tsx

```diff
@@ -0,0 +1,358 @@
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    border-radius: 6px;
+  }
+
+  ${props => {
+    if (!props.isHighContrast) {
+      return css`
+        &:after {
+          border-style: solid;
+          border-width: 1px;
+          border-color: ${props.on && !props.disabled ? get('colors.accent.fg') : get('colors.border.subtle')};
+          opacity: ${props.disabled ? '0.5' : props.on ? '0.2' : '1'};
+        }
+      `
+    } else {
+      return css`
+        border-style: solid;
+        border-color: ${props.on && !props.disabled ? get('colors.accent.fg') : get('colors.border.subtle')};
+        border-width: 1px;
+        outline-offset: 2px;
+
+        .Toggle-knob {
+          left: -1px;
+          top: -1px;
+          bottom: -1px;
+          transform: translateX(${props.on ? 'calc(100% + 1px)' : '0'});
+        }
+      `
+    }
+  }}
+
+  ${props => {
+    if (!props.disabled) {
+      if (props.on) {
+        return css`
+          &:hover,
+          &:focus:focus-visible {
+            :after {
+              background-color: ${get('colors.accent.muted')};
+              opacity: 0.2;
+            }
+          }
+
+          &:active,
+          &:active:focus-visible {
+            :after {
+              background-color: ${get('colors.accent.muted')};
+              opacity: 0.5;
+            }
+          }
+        `
+      } else {
+        return css`
+          &:hover,
+          &:focus:focus-visible {
+            .Toggle-knob {
+              background-color: ${get('colors.btn.hoverBg')};
+              border-color: ${get('colors.border.default')};
+            }
+          }
+
+          &:active,
+          &:active:focus-visible {
+            .Toggle-knob {
+              background-color: ${get('colors.btn.activeBg')};
+            }
+          }
+        `
+      }
+    }
+  }}
+
+  &:disabled {
+    cursor: not-allowed;
+  }
+
+  ${sx}
+  ${sizeVariants}
+`
+
+const hiddenTextStyles: BetterSystemStyleObject = {
+  visibility: 'hidden',
+  height: 0
+}
+
+const Switch: React.FC<SwitchProps> = ({
+  'aria-labelledby': ariaLabelledby,
+  'aria-describedby': ariaDescribedby,
+  defaultOn,
+  disabled,
+  isLoading,
+  on: onProp,
+  onChange,
+  onClick,
+  size,
+  statusLabelPosition,
+  sx: sxProp
+}) => {
+  const isControlled = typeof defaultOn === 'undefined' && typeof onProp !== 'undefined'
+  const {theme, colorScheme} = useTheme()
+  const [onState, setOnState] = useProvidedStateOrCreate<boolean>(onProp, onChange, Boolean(defaultOn))
+  const acceptsInteraction = !disabled && !isLoading
+  const handleToggleClick: MouseEventHandler = useCallback(
+    e => {
+      if (!isControlled) {
+        setOnState(!onState)
+      }
+      onClick && onClick(e)
+    },
+    [onClick, isControlled, onState, setOnState]
+  )
+
+  useEffect(() => {
+    if (onChange && isControlled) {
+      onChange(Boolean(onProp))
+    }
+  }, [onChange, onProp, isControlled])
+
+  return (
+    <Box
```

Comment:
Should this component render a visually hidden checkbox in case people plan to use this inside a form?

Reply:
We intentionally used a button with `role="switch"` because this component _should not_ be used in a form. The accessibility team doesn't want checkbox changes to be saved unless the user explicitly clicks a "save" or "submit" button.

cc @alliethu or @ichelsea to keep me correct

Reply:
Ah okay. Should we mention that in the docs?

Reply:
`<button role=switch>` sounds like the right thing to do.

Reply:
Sure, I'll include something in the docs.

Reply:
@colebemis - I'm not sure where in the React docs this should go. Could I add a new heading for accessibility?